pkg/internal/checkup/checkup.go

```diff
@@ -154,6 +154,10 @@ func (c *Checkup) Run(ctx context.Context) error {
 			c.results.TrafficGenSentPackets, c.results.VMUnderTestReceivedPackets)
 	}
 
+	if c.results.TrafficGenSentPackets == 0 {
+		return fmt.Errorf("no packets were generated from the traffic generator")
+	}
```

Comment:
Could you please move this check to be the first?
If there were no generated packets, there is no need to continue checking the other conditions.